File: 01 Introdução/3301 Sobrinho do Meio/3301.py
H, Z, L = [int(x) for x in input().split(" ")]
if H > Z and H < L or H > L and H < Z: print("huguinho")
if L > Z and L < H or L > H and L < Z: print("luisinho")
if Z > H and Z < L or Z > L and Z < H: print("zezinho")

# https://www.beecrowd.com.br/judge/pt/problems/view/3301
# Ranking 00252º, Time 0.039, Submiss Date 12/04/2023 13:48:51 (Python 3.9)

File: 01 Introdução/3301 Sobrinho do Meio/3301.cpp
# include <iostream>

using namespace std;

int main(){
  int A, B, C;
  cin >> A >> B >> C;
  if ((A > B && A < C) || (A > C && A < B)){
    cout << "huguinho" << endl;
  } else if ((B > A && B < C) || (B > C && B < A)){
    cout << "zezinho" << endl;
  } else {
    cout << "luisinho" << endl;
  }

  return 0;
}

// https://www.beecrowd.com.br/judge/pt/problems/view/3301
// Ranking 00412º, Time 0.000, Submiss Date 16/04/2023 17:08:14 (C++17)